pkgs/native_toolchain_c/test/clinker/build_testfiles.dart

```diff
@@ -16,8 +16,11 @@ Future<Uri> buildTestArchive(
   OS targetOS,
   Architecture architecture, {
   int? androidTargetNdkApi, // Must be specified iff targetOS is OS.android.
+  int? macOSTargetVersion, // Must be specified iff targetOS is OS.macos.
 }) async {
   assert((targetOS != OS.android) == (androidTargetNdkApi == null));
+  assert((targetOS != OS.macOS) == (macOSTargetVersion == null));
```

Comment:
Should we consider throwing argument errors instead of doing asserts?

Especially because below we check `macOSTargetVersion != null` instead of `targetOS == OS.macOS`.